### ISSUE TITLE
tabSlideBox.js line 134 reports undefined is not a function

### ISSUE THREAD

**Issue by OpenNoob** [saravmajestic/ionic#14]
Hi,

Line 134 

$timeout(function(){$scope.onSlideMove({"index" : eval(index)});},100);

reports undefined. I commented it out but not sure if it's doing something very crucial.

Please advise. Thanks.

**Comment by Durss** [comment]
If you declare an "onSlideMove" method on your controler's $scope you won't get the error neither.

Actually this method should check if a method "onSlideMove" has been defined on the scope before trying to call it.

Commenting it won't break anything but you will loose the possibility to capture the tab changing event.
Not a big deal if you don't need it.

Well, at least that's what i understood of the whole thing after some minutes of investigation to slove this same issue on my side.


Basically, this solves the issue in a cleaner way :
```javascript
angular.module('test', [])

.controller('TestCtrl', function($scope) {

	$scope.onSlideMove = function(data){
		//Just do nothing here if you don't care !
		console.log("You have selected " + data.index + " tab");
	};
});
```

**Comment by Durss** [comment]
I made a Pull Request to fix this :
https://github.com/saravmajestic/ionic/pull/15

My first one, i hope i made it correctly ! Apologies if that's not the case ^^

**Comment by Deep2322** [comment]
the height of the tab remains same,if tab 'A's height is 100px and tab 'B's height is 500px.if i move to tab 'B' and come back to tab 'A'.the height remains same as tab 'B's height .How to have an auto height for tabs
